ansible_base/resource_registry/tasks/sync.py

```diff
@@ -148,22 +148,105 @@ def get_resource_type_names() -> list[str]:
+    # How might we end up with a conflict?
+    # - get_orphan_resources ignores resources that are owned by the service and have
+    #   is partially migrated set to false. We could be conflicting with one of those.
+    # - You could rename A --> B and B --> C. If A is updated before B is, then we will
+    #   end up needing to update B before we can update A.
+    # - You could rename A --> B and B --> A. If this happens, there is really no recovery.
+    #   Updating either would result in a conflict. We would have to set A or B to some
+    #   temporary value. Hopefully this will never happen since this scenario would be
+    #   incredibly difficult to pull off.
+
+    conflict_resource = resource_type.get_conflicting_resource(resource_data)
+
+    if conflict_resource is None:
+        return
+
+    resp = api_client.get_resource(conflict_resource.ansible_id)
+
+    # If the conflicting resource doesn't exist on the server, just go ahead and delete it.
+    # This will most likely happen with resources that are ignored by get_orphan_resources.
+    if resp.status_code == 404:
+        delete_resource(conflict_resource)
+
+    # If the resource does exist, lets update it first. Hopefully this desn't also result
+    # in a duplicate key error. If it does, we're cooked.
+    elif resp.status_code == 200:
+        data = resp.json()
+        conflict_resource.update_resource(
+            resource_data=data["resource_data"], service_id=data["service_id"], is_partially_migrated=data["is_partially_migrated"]
+        )
+    else:
+        raise ResourceSyncHTTPError
+
+
 def _attempt_update_resource(
     manifest_item: ManifestItem,
     resource: Resource,
     resource_data: dict,
+    api_client: ResourceAPIClient,
     **kwargs,
 ) -> SyncResult:
     """Try to update existing resource."""
     try:
         resource.update_resource(resource_data, partial=True, **kwargs)
     except IntegrityError:  # pragma: no cover
-        return SyncResult(SyncStatus.CONFLICT, manifest_item)
+        # This typically means that there was a duplicate key error. To mitigate this
+        # we will attempt to hanlde the conflicting resource and perform the operation
+        # again.
+        try:
+            _handle_conflict(resource_data, resource.resource_type_obj, api_client)
+            resource.update_resource(resource_data, partial=True, **kwargs)
+        except (ResourceDeletionError, IntegrityError) as e:
+            logger.error(f"Failed to gracefully handle conflict for {resource_data}. Got error {e}.")
+            return SyncResult(SyncStatus.CONFLICT, manifest_item)
     except Error as e:
+        # Something happened with the database. We don't know what it is. Instead of failing the whole
+        # sync, we'll raise an error and skip this for now.
         logger.error(f"Failed to update resource {resource.ansible_id}. Received error: {e}")
```

Comment:
Can we alter this message to say something like "Will try again on the nest sync"?